packages/framework/src/Twig/RequiredSettingExtension.php

```diff
@@ -0,0 +1,235 @@
+use Shopsys\FrameworkBundle\Model\Product\Unit\UnitFacade;
+use Shopsys\FrameworkBundle\Model\Stock\StockFacade;
+use Symfony\Component\Routing\RouterInterface;
+use Twig\Environment;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class RequiredSettingExtension extends AbstractExtension
+{
+    /**
+     * @param \Twig\Environment $twig
+     * @param \Symfony\Component\Routing\RouterInterface $router
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Component\Setting\Setting $setting
+     * @param \Shopsys\FrameworkBundle\Model\Mail\MailTemplateFacade $mailTemplateFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterFacade $parameterFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Unit\UnitFacade $unitFacade
+     * @param \Shopsys\FrameworkBundle\Model\Stock\StockFacade $stockFacade
+     * @param \Shopsys\FrameworkBundle\Model\Country\CountryFacade $countryFacade
+     */
+    public function __construct(
+        protected readonly Environment $twig,
+        protected readonly RouterInterface $router,
+        protected readonly Domain $domain,
+        protected readonly Setting $setting,
+        protected readonly MailTemplateFacade $mailTemplateFacade,
+        protected readonly ParameterFacade $parameterFacade,
+        protected readonly UnitFacade $unitFacade,
+        protected readonly StockFacade $stockFacade,
+        protected readonly CountryFacade $countryFacade,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('render_required_settings', $this->renderRequiredSettings(...), ['is_safe' => ['html']]),
+        ];
+    }
+
+    /**
+     * @return string|null
+     */
+    public function renderRequiredSettings(): ?string
+    {
+        $requiredSettingsMessages = [];
+
+        $this->checkEnabledMailTemplatesHaveTheirBodyAndSubjectFilled($requiredSettingsMessages);
+        $this->checkAtLeastOneUnitExists($requiredSettingsMessages);
+        $this->checkDefaultUnitIsSet($requiredSettingsMessages);
+        $this->checkAtLeastOneWarehouseExists($requiredSettingsMessages);
+        $this->checkAtLeastOneCountryExists($requiredSettingsMessages);
+        $this->checkMandatoryArticlesExist($requiredSettingsMessages);
+        $this->checkAllSliderNumericValuesAreSet($requiredSettingsMessages);
+
+        if (count($requiredSettingsMessages) === 0) {
+            return null;
+        }
+
+        return $this->twig->render(
+            '@ShopsysFramework/Components/RequiredSettings/requiredSettings.html.twig',
+            [
+                'requiredSettingsMessages' => $requiredSettingsMessages,
+            ],
+        );
+    }
+
+    /**
+     * @param string[] $warnings
+     */
+    protected function checkEnabledMailTemplatesHaveTheirBodyAndSubjectFilled(array &$warnings): void
```

Comment:
instead of using the references, I would rather make `requiredSettingsMessages` a class property, however, it is a matter of taste so I can live with this solution as well :slightly_smiling_face:

Reply:
What was I thinking? Of course the class property is a better solution 😇

fixed in c3bbb989a3